[Chen Fan] iP

CHI-ME opened the pull request:
### Ninon

> “Your mind is for having ideas, not holding them.” – David [Allen](https://dansilvestre.com/productivity-quotes)

Ninon frees your mind of having to remember things you need to do. It's,


- text-based
- easy to learn
-  ~~FAST~~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://github.com/CHI-ME/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing todos
- [x] Managing deadlines
- [x] Managing events

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

`public static void main(String[] args) {
        new Ninon("tasks.txt").run();
    }`

CHI-ME committed:
rename, add greeting and exit

CHI-ME committed:
add echo function

CHI-ME committed:
add add_List() and print_List() function

CHI-ME committed:
add two class Task and TaskList

CHI-ME committed:
implement classes Todo, Deadline, and Event to inherit from Task

CHI-ME committed:
add a common used exception class DukeException

CHI-ME committed:
add delete function in TaskList class

CHI-ME committed:
add save tasks to disk function

CHI-ME committed:
Merge branch 'branch-Level-7'

CHI-ME committed:
store input of date in jave.time.LocalDate instead of String

CHI-ME committed:
Merge branch 'branch-Level-8'

CHI-ME committed:
make the project more OOP (still bug on export data)

CHI-ME committed:
fixed IO issues

CHI-ME committed:
Organize all the classes into Duke packages

CHI-ME committed:
add gradle support

NoahKoh commented on src/main/java/Duke/MarkCommand.java:
```diff
@@ -0,0 +1,24 @@
+package Duke;
+
+public class MarkCommand extends Command {
+    public int num;
```
Maybe can use something more descriptive like index or taskIndex.

NoahKoh commented on src/main/java/Duke/MarkCommand.java:
```diff
@@ -0,0 +1,24 @@
+
+public class MarkCommand extends Command {
+    public int num;
+    public boolean is_mark;
```
Perhaps camelCase isMark or is_Mark would be better.

NoahKoh commented on src/main/java/Duke/DeleteCommand.java:
```diff
@@ -0,0 +1,18 @@
+package Duke;
+
+public class DeleteCommand extends Command {
+    public int num;
```
Maybe can use something more descriptive like index or taskIndex.

NoahKoh commented on src/main/java/Duke/Ninon.java:
```diff
@@ -0,0 +1,42 @@
+    }
+    public void run() {
+        ui.greeting();
+        boolean isExit = false;
```
Boolean sounds like boolean, good

NoahKoh commented on src/main/java/Duke/Ninon.java:
```diff
@@ -0,0 +1,42 @@
+package Duke;
```
package name should all be in lowercase

NoahKoh commented on src/main/java/Duke/Task.java:
```diff
@@ -0,0 +1,32 @@
+package Duke;
+
+public class Task {
+    private Boolean done;
```
I think the boolean type should be lowercase, and the name should sound like a boolean like isDone.

NoahKoh reviewed (commented):
Overall, I see your code follows the Java coding standards and naming convetion except for some minor issue. LGTM

CHI-ME committed:
Add few JUnit Tests

CHI-ME committed:
create a jar file

CHI-ME committed:
add javadoc comment to Command and Task subclasses

CHI-ME committed:
Merge branch 'branch-A-JavaDoc'

CHI-ME committed:
Tweak the code to comply with a coding standard

CHI-ME committed:
Merge branch 'branch-A-CodingStandard'

CHI-ME committed:
add find method

CHI-ME committed:
Merge branch 'branch-Level-9'

lyhthaddeus commented on src/main/java/Ninon/AddCommand.java:
```diff
@@ -0,0 +1,43 @@
+/**
+ * Represents a command to add a task to the task list.
+ */
+public class AddCommand extends Command {
```
Could Command all be placed in one package?

lyhthaddeus commented on src/main/java/Ninon/Command.java:
```diff
@@ -0,0 +1,24 @@
+ * Represents an abstract command that can be executed on a task list.
+ * Subclasses should implement specific behaviors for different types of commands.
+ */
+public abstract class Command {
```
I like to use of abstract class to encapsulate the idea of a command that cant be initialized.

lyhthaddeus commented on src/main/java/Ninon/Command.java:
```diff
@@ -0,0 +1,24 @@
+     *
+     * @return true if the command is an exit command, false otherwise.
+     */
+    public abstract boolean isExit();
```
Does this need to be abstract since you only have 1 exit command (which can override this method)? 
```suggestion
public boolean isExit() {
   return false;
}
```

lyhthaddeus commented on src/main/java/Ninon/DeleteCommand.java:
```diff
@@ -0,0 +1,43 @@
+    /**
+     * The index of the task to be deleted.
+     */
+    public int num;
```
is 'num' a really clear name for a variable, perhaps 'Id' could be considered?

lyhthaddeus commented on src/main/java/Ninon/Parser.java:
```diff
@@ -0,0 +1,67 @@
+
+    public static Command parse(String message) throws DukeException {
+        Command command;
+        if (Objects.equals(message, "bye")) {
```
could a switch case with enum be used here?

lyhthaddeus reviewed (commented):
Overall nice code, easy to read and follow. Usage of package could be more in the code
to help with easier navigation

CHI-ME committed:
Add GUI feature

CHI-ME committed:
Add assert feature

There are few important assumptions in class Parser remain unhandled

To ensure the input is correctly recognized, few assertions are added to
the class Parser and TaskList

CHI-ME committed:
Merge pull request #2 from CHI-ME/branch-A-Assertions

Add assert feature

CHI-ME committed:
Merge branch 'master' into branch-A-CodeQuality

CHI-ME committed:
Improve code quality and structure

CHI-ME committed:
Merge pull request #3 from CHI-ME/branch-A-CodeQuality

Branch a code quality

CHI-ME committed:
Add B-DoAfterTasks extension

CHI-ME committed:
Update image of Ui, user and bot

CHI-ME committed:
Update README.md

CHI-ME committed:
Update README.md

CHI-ME committed:
Fix bugs in class Parser

CHI-ME committed:
Update README.md

CHI-ME committed:
Fix bugs in class Parser

CHI-ME committed:
Update README.md

CHI-ME committed:
Update Ui.png
